adapters/stream.py

```diff
@@ -0,0 +1,96 @@
+#  -*- coding: utf-8 -*-
+# *********************************************************************
+# plankton - a library for creating hardware device simulators
+# Copyright (C) 2016 European Spallation Source ERIC
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# *********************************************************************
+
+
+import asyncore
+import asynchat
+import socket
+
+from adapters import Adapter
+from datetime import datetime
+
+
+class StreamHandler(asynchat.async_chat):
+    def __init__(self, sock, target, bindings):
+        asynchat.async_chat.__init__(self, sock=sock)
+        self.set_terminator(bindings['meta']['in_terminator'])
+        self.target = target
+        self.bindings = bindings
+        self.buffer = []
+
+    def collect_incoming_data(self, data):
+        self.buffer.append(data)
+
+    def found_terminator(self):
+        request = ''.join(self.buffer)
+        reply = None
+        self.buffer = []
+
+        for command, funcname in self.bindings['commands'].iteritems():
+            if request.startswith(command):
+                func = getattr(self.target, funcname)
+                args = request[len(command):]
+                try:
+                    reply = func(args) if args else func()
+                except Exception:
```

Comment:
Maybe in this case there should be a comment why the exception is eaten. Normally I wouldn't consider that necessary but since there is this very specific reason I think it should be documented.